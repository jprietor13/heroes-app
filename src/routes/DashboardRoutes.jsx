import React from "react";
import { Route, Routes } from "react-router-dom";
import { Navbar } from "../components/ui/Navbar";
import { MarvelScreen } from "../components/marvel/MarvelScreen";
import { DcScreen } from "../components/dc/DcScreen";
import { SearchScreen } from "../components/search/SearchScreen";
import { HeroScreen } from "../components/hero/HeroScreen";

export const DashboardRoutes = () => {
  return (
    <>
      <Navbar />
      <div className="container mt-5">
      <Routes>
        <Route exact path="/" element={<MarvelScreen />} />
        <Route exact path="/marvel" element={<MarvelScreen />} />
        <Route exact path="/dc" element={<DcScreen />} />
        <Route exact path="/search" element={<SearchScreen />} />
        <Route exact path="/hero/:idHero" element={ <HeroScreen /> }/>
      </Routes>
      </div>
    </>
  );
};
