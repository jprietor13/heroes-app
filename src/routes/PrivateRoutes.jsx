import React, { useContext } from 'react'
import { AuthContext } from '../auth/authContext'
import { Navigate, useLocation } from 'react-router-dom'

export const PrivateRoutes = ({ children }) => {

  const { user } = useContext(AuthContext)

  const { search, pathname } = useLocation();

  localStorage.setItem('lastPath', pathname + search)

  return (
    user.logged ? children : <Navigate to="/login" /> 
  )
}
