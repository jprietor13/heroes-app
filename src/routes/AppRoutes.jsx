import React from 'react'
import { BrowserRouter, Routes, Route } from 'react-router-dom'
import { LoginScreen } from '../components/login/LoginScreen'
import { DashboardRoutes } from './DashboardRoutes'


export const AppRoutes = () => {
  return (
    <BrowserRouter>
      <Routes>
        <Route exact path="/login" element={ <LoginScreen /> }/>
        <Route exact path="*" element={ <DashboardRoutes /> } />
      </Routes>
    </BrowserRouter>
  )
}
