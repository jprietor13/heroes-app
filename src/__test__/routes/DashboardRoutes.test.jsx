import { mount } from "enzyme";
import { DashboardRoutes } from "../../routes/DashboardRoutes";
import { MemoryRouter } from "react-router-dom";
import { AuthContext } from "../../auth/authContext";

describe("Pruebas en <DashboardRoutes", () => {

  const valueContext = {
    user: {
      logged: false,
      name: "Juan Prieto",
    },
  };

  test("Debe de mostrarse correctamente - Marvel", () => {
    const wrapper = mount(
      <AuthContext.Provider value={valueContext}>
        <MemoryRouter initialEntries={ ['/'] }>
          <DashboardRoutes />
        </MemoryRouter>
      </AuthContext.Provider>
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.text-info').text().trim()).toBe('Juan Prieto')
  });

  test("Debe mostrarse correctamente - DC", () => {

    const wrapper = mount(
      <AuthContext.Provider value={valueContext}>
        <MemoryRouter initialEntries={ ['/dc'] }>
          <DashboardRoutes />
        </MemoryRouter>
      </AuthContext.Provider>
    )

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('h1').text().trim()).toBe('DC Heroes')

  })

});
