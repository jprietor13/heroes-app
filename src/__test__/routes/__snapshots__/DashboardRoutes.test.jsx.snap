// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <DashboardRoutes Debe de mostrarse correctamente - Marvel 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Asociaciones
    </a>
    <div
      className="navbar-collapse"
    >
      <div
        className="navbar-nav"
      >
        <a
          className="nav-item nav-link"
          href="/marvel"
          onClick={[Function]}
        >
          Marvel
        </a>
        <a
          className="nav-item nav-link"
          href="/dc"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          className="nav-item nav-link"
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
      </div>
    </div>
    <div
      className="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <span
          className="nav-item nav-link text-info"
        >
          Juan Prieto
        </span>
        <button
          className="nav-item nav-link btn"
          onClick={[Function]}
        >
          Logout
        </button>
      </ul>
    </div>
  </nav>,
  <div
    className="container mt-5"
  >
    Array [
      <h1>
        Marvel Heroes
      </h1>,
      <hr />,
      <div
        className="row rows-cols-1 row-cols-md-3 g-3 animate__animated animate__fadeIn"
      >
        <div
          className="col"
        >
          <div
            className="card"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-4"
              >
                <img
                  alt="Spider Man"
                  className="card-img-top"
                  src="/assets/images/marvel-spider.jpg"
                />
              </div>
              <div
                className="col-8"
              >
                <h4
                  className="card-title mt-2"
                >
                  Spider Man
                </h4>
                <p
                  className="card-text"
                >
                  Peter Parker
                </p>
                <p
                  className="card-text"
                >
                  Amazing Fantasy #15
                </p>
                <a
                  className=""
                  href="/hero/marvel-spider"
                  onClick={[Function]}
                >
                  Mas...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="card"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-4"
              >
                <img
                  alt="Captain America"
                  className="card-img-top"
                  src="/assets/images/marvel-captain.jpg"
                />
              </div>
              <div
                className="col-8"
              >
                <h4
                  className="card-title mt-2"
                >
                  Captain America
                </h4>
                <p
                  className="card-text"
                >
                  Steve Rogers
                </p>
                <p
                  className="card-text"
                >
                  Captain America Comics #1
                </p>
                <a
                  className=""
                  href="/hero/marvel-captain"
                  onClick={[Function]}
                >
                  Mas...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="card"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-4"
              >
                <img
                  alt="Iron Man"
                  className="card-img-top"
                  src="/assets/images/marvel-iron.jpg"
                />
              </div>
              <div
                className="col-8"
              >
                <h4
                  className="card-title mt-2"
                >
                  Iron Man
                </h4>
                <p
                  className="card-text"
                >
                  Tony Stark
                </p>
                <p
                  className="card-text"
                >
                  Tales of Suspense #39
                </p>
                <a
                  className=""
                  href="/hero/marvel-iron"
                  onClick={[Function]}
                >
                  Mas...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="card"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-4"
              >
                <img
                  alt="Thor"
                  className="card-img-top"
                  src="/assets/images/marvel-thor.jpg"
                />
              </div>
              <div
                className="col-8"
              >
                <h4
                  className="card-title mt-2"
                >
                  Thor
                </h4>
                <p
                  className="card-text"
                >
                  Thor Odinson
                </p>
                <p
                  className="card-text"
                >
                  Journey into Myster #83
                </p>
                <a
                  className=""
                  href="/hero/marvel-thor"
                  onClick={[Function]}
                >
                  Mas...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="card"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-4"
              >
                <img
                  alt="Hulk"
                  className="card-img-top"
                  src="/assets/images/marvel-hulk.jpg"
                />
              </div>
              <div
                className="col-8"
              >
                <h4
                  className="card-title mt-2"
                >
                  Hulk
                </h4>
                <p
                  className="card-text"
                >
                  Bruce Banner
                </p>
                <p
                  className="card-text"
                >
                  The Incredible Hulk #1
                </p>
                <a
                  className=""
                  href="/hero/marvel-hulk"
                  onClick={[Function]}
                >
                  Mas...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="card"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-4"
              >
                <img
                  alt="Wolverine"
                  className="card-img-top"
                  src="/assets/images/marvel-wolverine.jpg"
                />
              </div>
              <div
                className="col-8"
              >
                <h4
                  className="card-title mt-2"
                >
                  Wolverine
                </h4>
                <p
                  className="card-text"
                >
                  James Howlett
                </p>
                <p
                  className="card-text"
                >
                  The Incredible Hulk #180
                </p>
                <a
                  className=""
                  href="/hero/marvel-wolverine"
                  onClick={[Function]}
                >
                  Mas...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="card"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-4"
              >
                <img
                  alt="Daredevil"
                  className="card-img-top"
                  src="/assets/images/marvel-daredevil.jpg"
                />
              </div>
              <div
                className="col-8"
              >
                <h4
                  className="card-title mt-2"
                >
                  Daredevil
                </h4>
                <p
                  className="card-text"
                >
                  Matthew Michael Murdock
                </p>
                <p
                  className="card-text"
                >
                  Daredevil #1
                </p>
                <a
                  className=""
                  href="/hero/marvel-daredevil"
                  onClick={[Function]}
                >
                  Mas...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="card"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-4"
              >
                <img
                  alt="Hawkeye"
                  className="card-img-top"
                  src="/assets/images/marvel-hawkeye.jpg"
                />
              </div>
              <div
                className="col-8"
              >
                <h4
                  className="card-title mt-2"
                >
                  Hawkeye
                </h4>
                <p
                  className="card-text"
                >
                  Clinton Francis Barton
                </p>
                <p
                  className="card-text"
                >
                  Tales of Suspense #57
                </p>
                <a
                  className=""
                  href="/hero/marvel-hawkeye"
                  onClick={[Function]}
                >
                  Mas...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="card"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-4"
              >
                <img
                  alt="Cyclops"
                  className="card-img-top"
                  src="/assets/images/marvel-cyclops.jpg"
                />
              </div>
              <div
                className="col-8"
              >
                <h4
                  className="card-title mt-2"
                >
                  Cyclops
                </h4>
                <p
                  className="card-text"
                >
                  Scott Summers
                </p>
                <p
                  className="card-text"
                >
                  X-Men #1
                </p>
                <a
                  className=""
                  href="/hero/marvel-cyclops"
                  onClick={[Function]}
                >
                  Mas...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="card"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-4"
              >
                <img
                  alt="Silver Surfer"
                  className="card-img-top"
                  src="/assets/images/marvel-silver.jpg"
                />
              </div>
              <div
                className="col-8"
              >
                <h4
                  className="card-title mt-2"
                >
                  Silver Surfer
                </h4>
                <p
                  className="card-text"
                >
                  Norrin Radd
                </p>
                <p
                  className="card-text"
                >
                  The Fantastic Four #48
                </p>
                <a
                  className=""
                  href="/hero/marvel-silver"
                  onClick={[Function]}
                >
                  Mas...
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>,
    ]
  </div>,
]
`;

exports[`Pruebas en <DashboardRoutes Debe mostrarse correctamente - DC 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Asociaciones
    </a>
    <div
      className="navbar-collapse"
    >
      <div
        className="navbar-nav"
      >
        <a
          className="nav-item nav-link"
          href="/marvel"
          onClick={[Function]}
        >
          Marvel
        </a>
        <a
          aria-current="page"
          className="nav-item nav-link active"
          href="/dc"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          className="nav-item nav-link"
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
      </div>
    </div>
    <div
      className="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <span
          className="nav-item nav-link text-info"
        >
          Juan Prieto
        </span>
        <button
          className="nav-item nav-link btn"
          onClick={[Function]}
        >
          Logout
        </button>
      </ul>
    </div>
  </nav>,
  <div
    className="container mt-5"
  >
    Array [
      <h1>
        DC Heroes
      </h1>,
      <hr />,
      <div
        className="row rows-cols-1 row-cols-md-3 g-3 animate__animated animate__fadeIn"
      >
        <div
          className="col"
        >
          <div
            className="card"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-4"
              >
                <img
                  alt="Batman"
                  className="card-img-top"
                  src="/assets/images/dc-batman.jpg"
                />
              </div>
              <div
                className="col-8"
              >
                <h4
                  className="card-title mt-2"
                >
                  Batman
                </h4>
                <p
                  className="card-text"
                >
                  Bruce Wayne
                </p>
                <p
                  className="card-text"
                >
                  Detective Comics #27
                </p>
                <a
                  className=""
                  href="/hero/dc-batman"
                  onClick={[Function]}
                >
                  Mas...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="card"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-4"
              >
                <img
                  alt="Superman"
                  className="card-img-top"
                  src="/assets/images/dc-superman.jpg"
                />
              </div>
              <div
                className="col-8"
              >
                <h4
                  className="card-title mt-2"
                >
                  Superman
                </h4>
                <p
                  className="card-text"
                >
                  Kal-El
                </p>
                <p
                  className="card-text"
                >
                  Action Comics #1
                </p>
                <a
                  className=""
                  href="/hero/dc-superman"
                  onClick={[Function]}
                >
                  Mas...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="card"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-4"
              >
                <img
                  alt="Flash"
                  className="card-img-top"
                  src="/assets/images/dc-flash.jpg"
                />
              </div>
              <div
                className="col-8"
              >
                <h4
                  className="card-title mt-2"
                >
                  Flash
                </h4>
                <p
                  className="card-text"
                >
                  Jay Garrick
                </p>
                <p
                  className="text-muted"
                >
                  Jay Garrick, Barry Allen, Wally West, Bart Allen
                </p>
                <p
                  className="card-text"
                >
                  Flash Comics #1
                </p>
                <a
                  className=""
                  href="/hero/dc-flash"
                  onClick={[Function]}
                >
                  Mas...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="card"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-4"
              >
                <img
                  alt="Green Lantern"
                  className="card-img-top"
                  src="/assets/images/dc-green.jpg"
                />
              </div>
              <div
                className="col-8"
              >
                <h4
                  className="card-title mt-2"
                >
                  Green Lantern
                </h4>
                <p
                  className="card-text"
                >
                  Alan Scott
                </p>
                <p
                  className="text-muted"
                >
                  Alan Scott, Hal Jordan, Guy Gardner, John Stewart, Kyle Raynor, Jade, Sinestro, Simon Baz
                </p>
                <p
                  className="card-text"
                >
                  All-American Comics #16
                </p>
                <a
                  className=""
                  href="/hero/dc-green"
                  onClick={[Function]}
                >
                  Mas...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="card"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-4"
              >
                <img
                  alt="Green Arrow"
                  className="card-img-top"
                  src="/assets/images/dc-arrow.jpg"
                />
              </div>
              <div
                className="col-8"
              >
                <h4
                  className="card-title mt-2"
                >
                  Green Arrow
                </h4>
                <p
                  className="card-text"
                >
                  Oliver Queen
                </p>
                <p
                  className="card-text"
                >
                  More Fun Comics #73
                </p>
                <a
                  className=""
                  href="/hero/dc-arrow"
                  onClick={[Function]}
                >
                  Mas...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="card"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-4"
              >
                <img
                  alt="Wonder Woman"
                  className="card-img-top"
                  src="/assets/images/dc-wonder.jpg"
                />
              </div>
              <div
                className="col-8"
              >
                <h4
                  className="card-title mt-2"
                >
                  Wonder Woman
                </h4>
                <p
                  className="card-text"
                >
                  Princess Diana
                </p>
                <p
                  className="card-text"
                >
                  All Star Comics #8
                </p>
                <a
                  className=""
                  href="/hero/dc-wonder"
                  onClick={[Function]}
                >
                  Mas...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="card"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-4"
              >
                <img
                  alt="Martian Manhunter"
                  className="card-img-top"
                  src="/assets/images/dc-martian.jpg"
                />
              </div>
              <div
                className="col-8"
              >
                <h4
                  className="card-title mt-2"
                >
                  Martian Manhunter
                </h4>
                <p
                  className="card-text"
                >
                  J'onn J'onzz
                </p>
                <p
                  className="text-muted"
                >
                  Martian Manhunter
                </p>
                <p
                  className="card-text"
                >
                  Detective Comics #225
                </p>
                <a
                  className=""
                  href="/hero/dc-martian"
                  onClick={[Function]}
                >
                  Mas...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="card"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-4"
              >
                <img
                  alt="Robin/Nightwing"
                  className="card-img-top"
                  src="/assets/images/dc-robin.jpg"
                />
              </div>
              <div
                className="col-8"
              >
                <h4
                  className="card-title mt-2"
                >
                  Robin/Nightwing
                </h4>
                <p
                  className="card-text"
                >
                  Dick Grayson
                </p>
                <p
                  className="card-text"
                >
                  Detective Comics #38
                </p>
                <a
                  className=""
                  href="/hero/dc-robin"
                  onClick={[Function]}
                >
                  Mas...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="card"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-4"
              >
                <img
                  alt="Blue Beetle"
                  className="card-img-top"
                  src="/assets/images/dc-blue.jpg"
                />
              </div>
              <div
                className="col-8"
              >
                <h4
                  className="card-title mt-2"
                >
                  Blue Beetle
                </h4>
                <p
                  className="card-text"
                >
                  Dan Garret
                </p>
                <p
                  className="text-muted"
                >
                  Dan Garret, Ted Kord, Jaime Reyes
                </p>
                <p
                  className="card-text"
                >
                  Mystery Men Comics #1
                </p>
                <a
                  className=""
                  href="/hero/dc-blue"
                  onClick={[Function]}
                >
                  Mas...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="card"
          >
            <div
              className="row no-gutters"
            >
              <div
                className="col-4"
              >
                <img
                  alt="Black Canary"
                  className="card-img-top"
                  src="/assets/images/dc-black.jpg"
                />
              </div>
              <div
                className="col-8"
              >
                <h4
                  className="card-title mt-2"
                >
                  Black Canary
                </h4>
                <p
                  className="card-text"
                >
                  Dinah Drake
                </p>
                <p
                  className="text-muted"
                >
                  Dinah Drake, Dinah Lance
                </p>
                <p
                  className="card-text"
                >
                  Flash Comics #86
                </p>
                <a
                  className=""
                  href="/hero/dc-black"
                  onClick={[Function]}
                >
                  Mas...
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>,
    ]
  </div>,
]
`;
